packages/salesforcedx-vscode-core/test/jest/commands/deployExecutor.test.ts

```diff
@@ -148,4 +161,75 @@ describe('Deploy Executor', () => {
     expect(ensureLocalTrackingSpy).not.toHaveBeenCalled();
     expect(deploySpy).toHaveBeenCalled();
   });
+
+  it('should clear errors on success', async () => {
+    const mockDeployResult = {
+      response: {
+        status: 'Succeeded'
+      }
+    };
+    const setPropertiesForFilesDeployMock = jest.fn();
+    const getInstanceSpy = jest
+      .spyOn(PersistentStorageService, 'getInstance')
+      .mockReturnValue({
+        setPropertiesForFilesDeploy: setPropertiesForFilesDeployMock
+      } as any);
+    const getPackageDirectoryPathsSpy = jest
+      .spyOn(SfdxPackageDirectories, 'getPackageDirectoryPaths')
+      .mockResolvedValue('path/to/foo' as any);
+    const createOutputSpy = jest
+      .spyOn(TestDeployExecutor.prototype as any, 'createOutput')
+      .mockReturnValue('path/to/foo' as any);
+    const appendLineSpy = jest
+      .spyOn(channelService, 'appendLine')
+      .mockImplementation(jest.fn());
+
+    DeployRetrieveExecutor.errorCollection = MockErrorCollection as any;
+    SfdxCommandletExecutor.errorCollection = MockErrorCollection as any;
+    const deployRetrieveExecutorClearSpy = jest.spyOn(
+      DeployRetrieveExecutor.errorCollection,
+      'clear'
+    );
+    const sfdxCommandletExecutorClearSpy = jest.spyOn(
+      SfdxCommandletExecutor.errorCollection,
+      'clear'
+    );
+
+    const executor = new TestDeployExecutor(
+      'testDeploy',
+      'force_source_deploy_with_sourcepath_beta'
+    );
+
+    // Act
+    await (executor as any).postOperation(mockDeployResult);
+
+    // Assert
+    expect(getInstanceSpy).toHaveBeenCalled();
+    expect(getPackageDirectoryPathsSpy).toHaveBeenCalled();
+    expect(createOutputSpy).toHaveBeenCalled();
+    expect(appendLineSpy).toHaveBeenCalled();
+    expect(setPropertiesForFilesDeployMock).toHaveBeenCalledWith(
+      mockDeployResult
+    );
+    expect(deployRetrieveExecutorClearSpy).toHaveBeenCalled();
+    expect(sfdxCommandletExecutorClearSpy).toHaveBeenCalled();
+  });
+
+  it('should unlock queue on failure', async () => {
+    // Arrange
+    const mockUnlock = jest.fn();
+    const unlockSpy = jest
+      .spyOn(DeployQueue, 'get')
+      .mockReturnValue({ unlock: mockUnlock } as any);
+    const executor = new TestDeployExecutor(
+      'testDeploy',
+      'force_source_deploy_with_sourcepath_beta'
+    );
```

Comment:
Where are we specifying that we have a failure while deploying in this test?

Reply:
we are not specifying/defining the result nor the success.. thus it's a failure